CMakeLists.txt

```
@@ -1,17 +1,12 @@
 # Icinga 2 | (c) 2012 Icinga GmbH | GPLv2+
 
-cmake_minimum_required(VERSION 2.8.12)
+# Funny but yes, pining this specific CMake version isn't due to AL2 but Ubuntu 20.04.
+cmake_minimum_required(VERSION 3.16.3)
```

> I've checked all the policies starting from [`CMake 3.0`](https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html#policies-introduced-by-cmake-3-0-removed-by-cmake-4-0) to [`CMake 3.17.0`](https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html#policies-introduced-by-cmake-3-17), so that should be all we need for the moment.

You should be able to specify this as well, we shouldn't actually have to raise the **required** version, `cmake_minimum_required` also allows to specify the supported policy version as well:

```cmake
cmake_minimum_required(VERSION 2.8.12...3.17)
```

Now raising the required version still makes sense to get rid of some old stuff no longer needed. Given that you remove `if("${CMAKE_VERSION}" VERSION_LESS "3.8")` (as I suggested), that would probably be sane required minimum as the other changes were in older versions (CMAKE_ENABLE_EXPORTS: 3.4; CMAKE_MACOSX_RPATH: 3.0). Hence, I'd suggest the following:

```suggestion
cmake_minimum_required(VERSION 3.8...3.17)
```

---

Why the burden of supporting policies for multiple versions? Who (ex. devs?) cares for v3.8..3.16.2? To me it's like C++/GCC – if you can require a higher version, you just do it.

---

Writing 3.8 here isn't a burden, it just describes the state of the CMake config with this PR more accurately. This doesn't mean we need to keep supporting 3.8 with other changes, so if at some point there's a desire to use a newer feature, it's still possible to raise the minimum accordingly.

---

> if at some point there's a desire to use a newer feature, it's still possible to raise the minimum accordingly.

That's the exact opposite of my way which is: raise ASAP – no need to do anything once a new feature is desired, but just use it.

Also, 3.8...3.17 confuses me, as well as a range at all where only a MINIMUM version is specified. It appears like there's some distro version which still requires v3.8.